src/core/packages/saved-objects/api-server-internal/src/lib/apis/internals/change_object_access_control.ts

```diff
@@ -0,0 +1,363 @@
+    ) {
+      const error = SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
+      return left({ id, type, error });
+    }
+
+    if (
+      authorizationResult.status !== 'fully_authorized' &&
+      !authorizationResult.typeMap.has(type)
+    ) {
+      const error = SavedObjectsErrorHelpers.decorateForbiddenError(
+        new Error(
+          `User is not authorized to ${
+            actionType === 'changeOwnership' ? 'change ownership' : 'change access mode'
+          } of type "${type}".`
+        )
+      );
+      return left({ id, type, error });
+    }
+
+    const currentSource = doc._source;
+
+    const versionProperties = getExpectedVersionProperties(
+      undefined,
+      doc as unknown as SavedObjectsRawDoc
+    );
+
+    const expectedResult = {
+      type,
+      id,
+      esRequestIndex: bulkOperationRequestIndexCounter++,
+    };
+
+    const documentMetadata = {
+      _id: serializer.generateRawId(undefined, type, id),
+      _index: getIndexForType(type),
+      ...versionProperties,
+    };
+
+    let documentToSave;
+    if (actionType === 'changeOwnership') {
+      documentToSave = {
+        updated_at: time,
+        updated_by: currentUserProfileUid,
+        accessControl: {
+          ...(currentSource?.accessControl || {}),
+          owner: newOwnerProfileUid,
+        },
+      };
+    } else {
+      const ownerFromSource = currentSource?.accessControl?.owner;
+      documentToSave = {
+        updated_at: time,
+        updated_by: currentUserProfileUid,
+        accessControl: {
+          owner: ownerFromSource,
+          accessMode: accessMode ?? 'default',
+        },
+      };
+    }
+    // @ts-expect-error BulkOperation.retry_on_conflict, BulkOperation.routing. BulkOperation.version, and BulkOperation.version_type are optional
+    bulkOperationParams.push({ update: documentMetadata }, { doc: documentToSave });
+
+    return right(expectedResult);
+  });
+
+  const bulkOperationResponse = bulkOperationParams.length
+    ? await client.bulk({
+        refresh: 'wait_for',
+        operations: bulkOperationParams,
+        require_alias: true,
+      })
+    : undefined;
+
+  return {
+    objects: expectedBulkOperationResults.map<{ id: string; type: string; error?: any }>(
+      (expectedResult) => {
+        if (isLeft(expectedResult)) {
+          return expectedResult.value;
+        }
+
+        const { type, id, esRequestIndex } = expectedResult.value;
+        const response = bulkOperationResponse?.items[esRequestIndex] ?? {};
+
+        const rawResponse = Object.values(
+          response
+        )[0] as unknown as GetBulkOperationErrorRawResponse;
+        const error = getBulkOperationError(type, id, rawResponse);
+        if (error) {
+          return { id, type, error };
+        }
+
+        return { id, type };
+      }
+    ),
+  };
+};
+
+function isMgetError(doc?: estypes.MgetResponseItem<unknown>): doc is estypes.MgetMultiGetError {
```

Comment:
`isMgetError` is already declared in `update_objects_spaces`. Let's reduce duplication and move the helper to the `internal_utils` file after `isMgetDoc`.